dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md

```diff
@@ -0,0 +1,105 @@
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form1.vb" id="increment_position_on_next_button_click":::
+
+## To check whether you've passed the end or beginning
+
+01. Create an event handler for the <xref:System.Windows.Forms.BindingSource.PositionChanged> event. In the handler, you can test whether the proposed position value has exceeded the actual data element count.
+
+     The following example illustrates how you can test whether you've reached the last data element. In the example, if you are at the last element, the **Next** button on the form is disabled.
+
+    > [!NOTE]
+    > Be aware that, should you change the list you are navigating in code, you should re-enable the **Next** button, so that users might browse the entire length of the new list. Additionally, be aware that the above <xref:System.Windows.Forms.BindingSource.PositionChanged> event for the specific <xref:System.Windows.Forms.BindingSource> you are working with needs to be associated with its event-handling method. The following is an example of a method for handling the <xref:System.Windows.Forms.BindingSource.PositionChanged> event:
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form1.cs" id="check_for_last_element":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form1.vb" id="check_for_last_element":::
+
+## To find an item and set it as the current item
+
+01. Find the record you wish to set as the current item. Use the <xref:System.Windows.Forms.BindingSource.Find%2A> method of the <xref:System.Windows.Forms.BindingSource>, if your data source implements <xref:System.ComponentModel.IBindingList>. Some examples of data sources that implement `IBindingList` are <xref:System.ComponentModel.BindingList%601> and <xref:System.Data.DataView>.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form1.cs" id="find_the_record":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form1.vb" id="find_the_record":::
+
+## How to ensure the selected row in a child table remains at the correct position
+
+When you work with data binding in Windows Forms, you'll display data in a parent/child or master/details view. It's a data-binding scenario where data from the same source is displayed in two controls. Changing the selection in one control causes the data displayed in the second control to change. For example, the first control might contain a list of customers and the second a list of orders related to the selected customer in the first control.
+
+When you display data in a parent/child view, you might have to take extra steps to ensure that the currently selected row in the child table isn't reset to the first row of the table. In order to do this, you'll have to cache the child table position and reset it after the parent table changes. Typically the child reset occurs the first time a field in a row of the parent table changes.
```

Comment:
Though it's understood for more clarity we can modify the sentence as:
**Typically,** the child **table** reset occurs the first time a field in a row of the parent table changes.